### PR TITLE
view renderers

### PR DESCRIPTION
**Pull request by tannernelson**
Related to:

- [x]  #534 (vapor/leaf#3)
- [x]  #521
- [x]  #503
- [x]  #466

Needs vapor/leaf#2

### CODE REVIEW

File: Sources/Vapor/Droplet/Droplet.swift [hunk 1]
```diff
@@ -242,6 +249,8 @@ public class Droplet {
             log.log(item.level, message: item.message)
         }
 
+        self.view = provided.view ?? LeafRenderer(viewsDir: workDir + "Resources/View")
```
Comment (LoganWright) [inline Sources/Vapor/Droplet/Droplet.swift]: Was this `Resources/Views` before? (plural)

Comment (tannernelson) [inline Sources/Vapor/Droplet/Droplet.swift]: Yes.

File: Sources/Vapor/View/LeafRenderer.swift [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import Leaf
+
+public final class LeafRenderer: ViewRenderer {
+    let stem: Stem
```
Comment (LoganWright) [inline Sources/Vapor/View/LeafRenderer.swift]: We should make this public. For people to add their own tags and stuff w/o needing a new object.

File: Sources/Vapor/View/View.swift [hunk 3]
```diff
@@ -1,50 +1,30 @@
...
-    ///Currently applied RenderDrivers
-    public static var renderers: [String: RenderDriver] = [:]
-
-    var data: Bytes
+public final class View {
+    public let data: Bytes
 
-    enum Error: Swift.Error {
-        case InvalidPath
+    public init(data: Bytes) {
```
Comment (LoganWright) [inline Sources/Vapor/View/View.swift]: should this be bytes so it can be `BytesConvertible`

File: Sources/Vapor/View/View.swift [hunk 4]
```diff
@@ -1,50 +1,30 @@
...
+    and from bytes for interfacing
+    with other byte based processes.
+*/
+extension View: BytesConvertible {
+    public func makeBytes() throws -> Bytes {
+        return data
     }
 
+    public convenience init(bytes: Bytes) throws {
+        self.init(data: bytes)
```
Comment (LoganWright) [inline Sources/Vapor/View/View.swift]: Is it worth having data too, or is just `bytes: ` enough.

Comment (tannernelson) [inline Sources/Vapor/View/View.swift]: The bytes one is throwing so there needs to be a different one.

**Pull request merged**